### DISCUSSION TITLE
Query for Java Static Initializer

### DISCUSSION THREAD

**Discussion by mrigankpawagi** [github/codeql#16838] [Q&A] [answered]
Hi, I am a junior developer and need some help with writing a query to capture the static initializer (`static { ... }`) of a Java class. Is this currently supported? If yes, could someone please point me to the right place in the documentation? Many thanks!

**Comment by jketema** [comment, marked as answer]
Hi,

In the database the static initialiser block becomes a block statement that is part of a static initialiser method. The easiest way I could find to get to it is as follows:
```ql
from Class c, StaticInitializer i, BlockStmt s
where c.getAMethod() = i and i.getBody().getAChild() = s
select s
```

**Reply by mrigankpawagi** [reply]
Thanks for your response! Let me try this and let you know :)

**Reply by mrigankpawagi** [reply]
Thank you, this worked!
